control-plane/catalog/to-consul/resource_test.go

```diff
@@ -56,6 +56,115 @@ func TestServiceResource_createDelete(t *testing.T) {
 	})
 }
 
+// Test that Loadbalancer service weight is set from service annotation.
+func TestServiceWeight_ingress(t *testing.T) {
+	t.Parallel()
+	client := fake.NewSimpleClientset()
+	syncer := newTestSyncer()
+	serviceResource := defaultServiceResource(client, syncer)
+
+	// Start the controller
+	closer := controller.TestControllerRun(&serviceResource)
+	defer closer()
+
+	// Insert an LB service
+	svc := lbService("foo", metav1.NamespaceDefault, "1.2.3.4")
+	svc.Annotations[annotationServiceWeight] = "22"
+	svc.Status.LoadBalancer.Ingress = append(
+		svc.Status.LoadBalancer.Ingress,
+		corev1.LoadBalancerIngress{IP: "3.3.3.3"},
+	)
+
+	svc.Status.LoadBalancer.Ingress = append(
+		svc.Status.LoadBalancer.Ingress,
+		corev1.LoadBalancerIngress{IP: "4.4.4.4"},
+	)
+
+	_, err := client.CoreV1().Services(metav1.NamespaceDefault).Create(context.Background(), svc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Verify what we got
+	retry.Run(t, func(r *retry.R) {
+		syncer.Lock()
+		defer syncer.Unlock()
+		actual := syncer.Registrations
+		require.Len(r, actual, 3)
+		require.Equal(r, "foo", actual[1].Service.Service)
+		require.Equal(r, "3.3.3.3", actual[1].Service.Address)
+		require.Equal(r, 22, actual[1].Service.Weights.Passing)
+		require.Equal(r, "foo", actual[2].Service.Service)
+		require.Equal(r, "4.4.4.4", actual[2].Service.Address)
+		require.Equal(r, 22, actual[2].Service.Weights.Passing)
+		require.NotEqual(r, actual[1].Service.ID, actual[2].Service.ID)
+	})
+}
+
+// Test that Loadbalancer service weight is set from service annotation.
+func TestServiceWeight_externalIP(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		Weight         string
+		ExtectedWeight int
+	}{
+		"external-IP": {
+			Weight:         "22",
+			ExtectedWeight: 22,
+		},
+		"non-int-weight": {
+			Weight:         "non-int",
+			ExtectedWeight: 0,
+		},
+		"one-weight": {
+			Weight:         "1",
+			ExtectedWeight: 0,
+		},
+		"zero-weight": {
+			Weight:         "0",
+			ExtectedWeight: 0,
+		},
+		"greater-than-100-is-allowed": {
+			Weight:         "1000",
+			ExtectedWeight: 1000,
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(tt *testing.T) {
+			client := fake.NewSimpleClientset()
+			syncer := newTestSyncer()
+			serviceResource := defaultServiceResource(client, syncer)
+
+			// Start the controller
+			closer := controller.TestControllerRun(&serviceResource)
+			defer closer()
+
+			// Insert an LB service
+			svc := lbService("foo", metav1.NamespaceDefault, "1.2.3.4")
```

Comment:
we can simplify this test. it looks like all we want to do is test the method `getSeviceWeight`. We don't need to configure the syncer. i think a much simpler test is directly invoking the method with different string values and verifying if the returned value is `-1, err` or `weight, nil` if that makes sense. happy to discuss this on a call!!

Reply:
Implemented as suggested 😃